contracts/periodic-prize-pool/PeriodicPrizePool.sol

```diff
@@ -332,32 +333,41 @@ abstract contract PeriodicPrizePool is Timelock, BaseRelayRecipient, ReentrancyG
 
   function redeemTicketsInstantlyFrom(
     address from,
-    uint256 tickets
-  ) 
-    external nonReentrant returns (uint256) 
+    uint256 tickets,
+    uint256 prepaidExitFee
+  )
+    external nonReentrant returns (uint256)
   {
     address operator = _msgSender();
+    require(__ticket.balanceOf(from) >= tickets, "PrizePool/insuff-tickets");
 
     uint256 userInterestRatioMantissa = _ticketInterestRatioMantissa(from);
-    uint256 exitFee = calculateExitFee(tickets, userInterestRatioMantissa);
+    uint256 totalExitFee = calculateExitFee(tickets, userInterestRatioMantissa);
 
-    // transfer the fee to this contract
-    _token().transferFrom(operator, address(this), exitFee);
+    uint256 userExitFee = totalExitFee;
+    uint256 sponsoredExitFee = (totalExitFee > prepaidExitFee) ? prepaidExitFee : totalExitFee;
+    if (totalExitFee > 0) {
+      userExitFee = (totalExitFee > prepaidExitFee) ? totalExitFee.sub(prepaidExitFee) : 0;
```

Comment:
This is a little confusing, as it's running the same condition `totalExitFee > prepaidExitFee` twice.

Since `sponsoredExitFee` represents the portion of the fee that is covered by the sponsor, it seems to me you can change lines `347`-`351` to:

```solidity
uint256 sponsoredExitFee = (totalExitFee > prepaidExitFee) ? prepaidExitFee : totalExitFee;
uint256 userExitFee = totalExitFee.sub(sponsoredExitFee);
```

Reply:
if totalExitFee is 0 we will get an underflow on `sub()`